python/hsfs/core/monitoring_window_config_engine.py

```diff
@@ -243,3 +249,193 @@ def get_window_start_end_times(
+            feature_name: str: Name of the feature to monitor.
+            start_time: int: Window start commit or event time
+            end_time: int: Window end commit or event time
+            row_percentage: fraction of rows to include [0, 1.0]
+            use_event_time: bool: Whether to use event time or ingestion time.
+                Feature View only.
+            transformation_function_dataset_version: int: Version of the dataset
+                to fetch statistics from for the transformation function.
+                Feature View only.
+
+        Returns:
+            `pyspark.sql.DataFrame`. A Spark DataFrame with the entity data
+        """
+        if entity.ENTITY_TYPE == "featuregroups":
+            # use_event_time and transformation_function don't apply here
+            entity_df = self.fetch_feature_group_data(
+                entity=entity,
+                feature_name=feature_name,
+                start_time=start_time,
+                end_time=end_time,
+            )
+
+        elif entity.ENTITY_TYPE == "featureview":
+            entity_df = self.fetch_feature_view_data(
+                entity=entity,
+                feature_name=feature_name,
+                start_time=start_time,
+                end_time=end_time,
+                use_event_time=use_event_time,
+                transformation_function_dataset_version=transformation_function_dataset_version,
+            )
+
+        if row_percentage < 1.0:
+            entity_df = entity_df.sample(fraction=row_percentage)
+
+        return entity_df
+
+    def fetch_feature_view_data(
+        self,
+        entity: "feature_view.FeatureView",
+        feature_name: Optional[str] = None,
+        start_time: Optional[int] = None,
+        end_time: Optional[int] = None,
+        use_event_time: bool = False,
+        transformation_function_dataset_version: Optional[int] = None,
+    ) -> TypeVar("pyspark.sql.DataFrame"):
+        """Fetch the feature view data based on time window and row percentage.
+
+        Args:
+            entity: FeatureView: Feature view to monitor.
+            feature_name: str: Name of the feature to monitor.
+            start_time: int: Window start commit or event time.
+            end_time: int: Window end commit or event time.
+            use_event_time: bool: Whether to use event time or ingestion time.
+            transformation_function_dataset_version: int: Dataset version of the transformation function
+
+        Returns:
+            `pyspark.sql.DataFrame`. A Spark DataFrame with the entity data
+        """
+        if use_event_time:
+            entity_df = entity._feature_view_engine.get_batch_query(
+                feature_view_obj=entity,
+                start_time=start_time,
+                end_time=end_time,
+                with_label=True if feature_name is None else False,
+                training_dataset_version=transformation_function_dataset_version,
```

Comment:
Have you checked what happens if training_dataset_version = None ?

Reply:
Ja, this version is used to add extra filters that could have been added when creating the training dataset. So if you pass none it just adds no extra filter